### ISSUE TITLE
Version 1.1.30 causes conflict with SixLabors.ImageSharp.Drawing

### ISSUE THREAD

**Issue by HakanL** [ststeiger/PdfSharpCore#98]
It seems that the SixLabors.Fonts project has been linked into this NuGet package, which then causes problems when someone needs to use SixLabors.ImageSharp.Drawing (which includes a reference to the SixLabors.Fonts project). The correct way should probably be to have a dependency on the SixLabors.Fonts package, not reference the csproj.

**Comment by ststeiger** [comment]
The issue is, the dependencies (Sixlabors.*) have been moved into git submodule. 
So after git clone, you need to run 
`git submodule update --init --recursive`
Then SixLabors.* is referenced as project. 

@startnow65 has probably removed the project references and added nuget references (and overlooked this one), because submodule dependencies apparently interfere with nuget for some reason, if only be that that they are not added automatically, or version conflicts.

**Comment by HakanL** [comment]
I understand, but when you generate NuGet packages you can't submodule other NuGet packages, since there will be a conflict of the same namespace from both NuGet packages with no way for the enduser to resolve it. The correct way is to specify the SixLabors.* dependencies as NuGet dependencies, not submodules.

**Comment by bdovaz** [comment]
Please release a new version updating the dependencies, the package you depend on now is deprecated and conflicts with the current one:

Deprecated package:

https://www.nuget.org/packages/SixLabors.Core/

Deprecation notice:
https://github.com/SixLabors/Core

Current package:

https://www.nuget.org/packages/SixLabors.ImageSharp/
https://github.com/SixLabors/ImageSharp

**It's really blocking us.**

**Comment by startnow65** [comment]
Sorry for the issues, would fix it today

**Comment by bdovaz** [comment]
@startnow65 including mine? Thanks.

**Comment by startnow65** [comment]
Yes @bdovaz , would try to switch back to using nugget packages for the dependencies instead of direct project reference.
This should also fix your issue.

In the mean time, i've unlisted version `1.1.30`

**Comment by bdovaz** [comment]
How is this going towards a Nuget release? I see a PR fixing this.

**Comment by ststeiger** [comment]
@bdovaz: Merged. 
Not that happy about removing the submodules, though ;)

**Comment by startnow65** [comment]
@ststeiger Sorry about that, i tried to make it work (for you) but it actually breaks a lot of things.
It does not build on my VS for example, but exact same thing builds with `dotnet build`.
Plus the issues with mixing and matching with other dependencies and runtime issues.

**Comment by ststeiger** [comment]
Well ok, we'll leave it at nuget for the future. 
Hmm, would be interesting to see if it builds on Rider then.

**Comment by bdovaz** [comment]
@ststeiger The most correct way in my opinion is as it is right now which is referencing the Nuget package.

Because if you reference the code directly through a git submodule like you were doing what happens if you later install the ImageSharp package from Nuget or if another package depends on ImageSharp? There would be conflicts from being all duplicated in different assemblies.

**Comment by ststeiger** [comment]
Only if you add the dependencies into the nuget package. 
As long as the assembly is called the same, there should be no conflicts. 
Unless you start to use strong names.  
Drawback then is that you need to add the (nuget or not) dependencies manually to your project.

**Comment by HakanL** [comment]
Most users (who use the PdfSharp NuGet package) add it via NuGet, not via sub modules, it's typically the recommended way to add dependencies. Also ImageSharp has strong names in their NuGet package, and that's exactly the error I'm getting (conflict on the strong name). I'm glad we're working on resolving this though!

**Comment by startnow65** [comment]
Hi @HakanL just in case you do not know already, version `1.2.0` has been released and should fix the reference issue you were having

**Comment by HakanL** [comment]
@startnow65 Perfect, thank you. I was just commenting on ststeiger's comment if you use NuGet for dependencies.

**Comment by ststeiger** [comment]
Wondering why anyone that is not developing a driver is using strong names, though. 
GAC is a thing of the past. 
This only makes sense with base framework versions where you need to be absolutely sure you have a specific version. 
Or when you digitally sign your application to prevent alteration for license enforcement.

**Comment by HakanL** [comment]
I don't know, but SixLabors are using it, so I don't think we have a lot of choices. That being said, I develop a lot of cloud solutions for ecommerce, etc, and I have never used submodules via git for any dependencies, but I use a lot of NuGet packages.
